src/bundles/pix_n_flix/index.ts

```diff
@@ -183,31 +195,180 @@ function loadMedia(): void {
   startVideo();
 }
 
+// just draws once on frame and stops video
+function snapPicture(): void {
+  drawFrame();
+  stopVideo();
+}
+
+// update fps
+function updateFPS(fps: number): void {
+  // prevent too big of an increase
+  if (fps < 2 || fps > 30) {
+    return;
+  }
+
+  const status = videoIsPlaying;
+  stopVideo();
+
+  FPS = fps;
+  setupData();
+
+  if (!status) {
+    setTimeout(() => snapPicture(), 50);
+    return;
+  }
+
+  startVideo();
+}
+
+// update the frame dimensions
+function updateDimensions(w: number, h: number): void {
+  if ((w === WIDTH && h === HEIGHT) || w > 500 || h > 500) {
+    return;
+  }
+
+  const status = videoIsPlaying;
+  stopVideo();
+
+  WIDTH = w;
+  HEIGHT = h;
+
+  videoElement.width = w;
+  videoElement.height = h;
+  canvasElement.width = w;
+  canvasElement.height = h;
+
+  setupData();
+
+  if (!status) {
+    setTimeout(() => snapPicture(), 50);
+    return;
+  }
+
+  startVideo();
+}
+
+let queue: Function = () => {};

```

Comment:
It is possible to be more specific about the type of this function insead of using the general type `Function`? From the usage in line 257, it seems to me that queue is of type `() => void`.